tests/test_json_schema_flattener.py

```diff
@@ -0,0 +1,308 @@
+            },
+            {
+                'type': 'object',
+                # No properties, required, or patternProperties
+                'additionalProperties': False,
+            },
+            {
+                'type': 'object',
+                'properties': {'b': {'type': 'integer'}},
+                # No required
+            },
+        ],
+    }
+
+    flattened = flatten_allof(copy.deepcopy(schema))
+    assert 'allOf' not in flattened
+    assert set(flattened['properties'].keys()) == {'a', 'b'}
+    assert flattened['required'] == ['a']  # Only from first member
+    assert flattened.get('additionalProperties') is False
+
+
+def test_flatten_allof_empty_properties_after_merge() -> None:
+    """Test edge case where properties/required/patternProperties might be empty."""
+    from pydantic_ai._json_schema import flatten_allof
+
+    schema: dict[str, Any] = {
+        'type': 'object',
+        'allOf': [
+            {
+                'type': 'object',
+                # No properties, required, or patternProperties
+            },
+        ],
+    }
+
+    flattened = flatten_allof(copy.deepcopy(schema))
+    assert 'allOf' not in flattened
+    # Should not have properties/required if they're empty
+    assert 'properties' not in flattened or not flattened.get('properties')
+    assert 'required' not in flattened or not flattened.get('required')
+
+
+def test_flatten_allof_with_initial_properties() -> None:
+    """Test flattening when root schema has initial properties (line 229)."""
+    from pydantic_ai._json_schema import flatten_allof
+
+    schema: dict[str, Any] = {
+        'type': 'object',
+        'properties': {'root_prop': {'type': 'string'}},  # Initial properties
+        'allOf': [
+            {
+                'type': 'object',
+                'properties': {'a': {'type': 'integer'}},
+                'required': ['a'],
+            },
+        ],
+    }
+
+    flattened = flatten_allof(copy.deepcopy(schema))
+    assert 'allOf' not in flattened
+    assert 'root_prop' in flattened['properties']
+    assert 'a' in flattened['properties']
+    assert flattened['required'] == ['a']
+
+
+def test_flatten_allof_with_pattern_properties() -> None:
+    """Test flattening when members have patternProperties (lines 241, 250)."""
+    from pydantic_ai._json_schema import flatten_allof
+
+    schema: dict[str, Any] = {
+        'type': 'object',
+        'allOf': [
+            {
+                'type': 'object',
+                'properties': {'a': {'type': 'string'}},
+                'patternProperties': {
+                    '^prefix_': {'type': 'string'},
+                },
+            },
+            {
+                'type': 'object',
+                'properties': {'b': {'type': 'integer'}},
+                'patternProperties': {
+                    '^suffix_': {'type': 'number'},
+                },
+            },
+        ],
+    }
+
+    flattened = flatten_allof(copy.deepcopy(schema))
```

Comment:
I'd prefer to not use the function directly, only the transformer classes.

And as mentioned above: please use a single `snapshot()` assertion (using the inline-snapshot library) rather than many asserts.